pkg/exporter/exporter.go

```diff
@@ -43,58 +46,66 @@ func New(clusterClient clientset.Interface, apiClient *client.APIClient, recorde
 }
 
 func (e *Exporter) Run(ctx context.Context, configMapName, region string, patrolInterval time.Duration, stopChan <-chan struct{}) {
+	//delete old configmap if any
+	err := e.DeleteConfigMap(ctx, configMapName)
+	if err != nil && !apierrors.IsNotFound(err) {
+		return
+	}
 	// create configMap first time
-	err := e.RefreshData(ctx, configMapName, region, stopChan)
+	err = e.RefreshData(ctx, configMapName, region)
 	if err != nil {
 		return
 	}
-	configMapWatch := e.GetGonfigMapWatch(ctx, configMapName)
 
-	// configMapWatch is reassigned when the config map is deleted,
-	// so we want to make sure to stop the last instance.
-	defer func() {
-		configMapWatch.Stop()
-	}()
+	informerFactory := informers.NewSharedInformerFactory(e.clusterClient, time.Hour*1)
+	configMapInformer := informerFactory.Core().V1().ConfigMaps().Informer()
+
+	// Create a channel to receive events from the informer
+	eventChan := make(chan interface{})
+	defer close(eventChan)
+
+	configMapInformer.AddEventHandler(
+		cache.ResourceEventHandlerFuncs{
+			DeleteFunc: func(obj interface{}) {
+				eventChan <- obj
+			},
+		})
+
+	go configMapInformer.Run(stopChan)
+	// Wait for the informer to sync
+	if !cache.WaitForCacheSync(stopChan, configMapInformer.HasSynced) {
+		runtime.HandleError(fmt.Errorf("timed out waiting for cache to sync"))
+		return
+	}
 
 	refreshPatrol := time.NewTicker(patrolInterval)
 	defer refreshPatrol.Stop()
 
 	for {
 		select {
-		// if the configMap got deleted by user
-		case event := <-configMapWatch.ResultChan():
-			if event.Type == watch.Deleted {
-				err := e.RefreshData(ctx, configMapName, region, stopChan)
-				if err != nil {
-					return
-				}
-				// refresh watch after deletion
-				configMapWatch.Stop()
-				configMapWatch = e.GetGonfigMapWatch(ctx, configMapName)
-
-				e.recorder.Eventf(&corev1.ObjectReference{
-					Kind:      "Pod",
-					Namespace: client.Namespace,
-					Name:      client.PodName,
-				}, corev1.EventTypeWarning, "Configmap Deleted", "Configmap got deleted")
-			}
-
-		// if refresh time elapsed
-		case <-refreshPatrol.C:
-			err := e.RefreshData(ctx, configMapName, region, stopChan)
+		case <-eventChan:
+			err := e.RefreshData(ctx, configMapName, region)
```

Comment:
You need to check the object and make sure it is the target configmap. Otherwise, every cfgmap deletion will trigger data refresh. Note that this informer is a global informer that watches all cfgmaps.

Reply:
Added.